pkg/daemon/rpm-ostree.go

```diff
@@ -185,11 +193,176 @@ func (r *RpmOstreeClient) RebaseLayered(imgURL string) error {
+		// a previous killed MCD (unlikely, but possible)
+		if err := cleanupTemporalOstreePolicyFileDropin(); err != nil {
+			klog.Errorf("Error deleting temporary MCD temporal policy %v", err)
+		}
+	}()
+	// Temporary patch the containers policies to allow rpm-ostree to pull
+	// an image from local storage. Only required if the policies are
+	// restrictive and won't allow containers-storage transport pulls.
+	if err := patchPoliciesForContainerStorage(podmanImageInfo); err != nil {
+		// Swallow the error and let it fail in case the user/default defined policies
+		// avoids pulling the image
+		klog.Errorf("Error writing temporal policy files %v", err)
+	}
+
+	klog.Infof("Executing local container storage rebase to %s", podmanImageInfo.RepoDigest)
+	return runRpmOstree("rebase", "--experimental", "ostree-unverified-image:containers-storage:"+podmanImageInfo.RepoDigest)
+}
+
+func restoreTemporalPolicyFile() error {
+	err := os.Rename(imagePolicyFilePath+".mcd.bak", imagePolicyFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	return nil
+}
+
+// patchPoliciesForContainerStorage temporarily overrides the container image policy visible
+// to rpm-ostreed to ensure pulls from the "containers-storage" transport are allowed for the
+// given image.
+//
+// This is necessary for tools like rpm-ostree to function correctly with locally
+// stored images, especially in environments with restrictive security policies. A common
+// scenario is a user removing the default "insecureAcceptAnything" policy without
+// adding an explicit rule for local storage, which is an easily missed implementation detail.
+//
+// The function is idempotent and will not modify the policy if it's already permissive
+// enough for local storage pulls.
+//
+// To avoid modifying the system's policy file, this function creates a temporary policy file
+// at rpmOstreeTemporalPolicyFile and uses a systemd drop-in to bind-mount it over the actual
+// policy file for the rpm-ostreed service. The drop-in and temporary policy file are cleaned
+// up after the rpm-ostree operation completes.
+func patchPoliciesForContainerStorage(podmanImageInfo *PodmanImageInfo) error {
+	url, err := generateTransportPolicyKeyForReference(podmanImageInfo)
+	if err != nil {
+		return err
+	}
+	_, err = os.Stat(imagePolicyFilePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			klog.Warningf("No policy file found at %s. Skipping temporal policy generation.", imagePolicyFilePath)
+			return nil
+		}
+		return err
+	}
+
+	policyOriginalContent, err := os.ReadFile(imagePolicyFilePath)
+	if err != nil {
+		return err
+	}
+
+	policy, err := signature.NewPolicyFromBytes(policyOriginalContent)
+	if err != nil {
+		return err
+	}
+
+	_, containerStoragePoliciesPresent := policy.Transports[imagePolicyTransportContainerStorage]
+	if (reflect.DeepEqual(policy.Default[0], signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()}) && !containerStoragePoliciesPresent) {
```

Comment:
Curious, can there be other fields in the `policy.Default[0]` if we have insecureAcceptAnything?